user_guide/src/missing_data.md

```diff
@@ -0,0 +1,171 @@
+# Missing data
+
+This page sets out how missing data is represented in `Polars` and how missing data can be filled.
+
+## `null` and `NaN` values
+
+Each `Series` or column in a `DataFrame` is an Arrow array based on the Apache Arrow format.
+Missing data is represented in Arrow and `Polars` with a `null` value. This `null` missing value applies for all data types including numerical values.
+
+`Polars` also allows `NotaNumber` or `NaN` values for float columns. These `NaN` values are considered to be a type of floating point data rather than missing data. We discuss `NaN` values separately below.
+
+You can manually define a missing value with the python `None` value:
+
+```python
+{{#include examples/missing_data/missing_types.py:4:4}}
+print(df)
+```
+
+```text
+{{#include outputs/missing_data/none_missing_value_df.txt}}
+```
+
+> In `Pandas` the value for missing data depends on the dtype of the column. In `Polars` missing data is always represented as a `null` value.
+
+## Missing data metadata
+
+The Arrow arrays used by `Polars` for each column in a `DataFrame` store metadata related to missing data.
+
+Firstly, there is the `null_count`: this is the number of rows with `null` values in the column.
+
+Secondly, if the `null_count` is greater than 0 there is a *validity bitmap*. The validity bitmap is an array that tells you whether each value is present or missing. The validity bitmap is part of the Arrow array for each column.
+
+You can access the `null_count` for a `DataFrame` with the `null_count` method:
+
+```python
+{{#include examples/missing_data/missing_types.py:6:6}}
+print(nullCountDf)
+```
+
+```text
+{{#include outputs/missing_data/null_count_df.txt}}
+```
+
+Although `null_count` is a method in python-polars this is a cheap operation as `null_count` is already calculated for the underlying Arrow array for that column.
+
+You can access a `Series` based on the validity bitmap for a column with the `is_null` method:
+
+```python
+{{#include examples/missing_data/missing_types.py:8:8}}
+print(isNullSeries)
+```
+
+```text
+{{#include outputs/missing_data/isnull_series.txt}}
+```
+
+As for `null_count` the `is_null` method is a cheap operation as the validity bitmap is already calculated for the Arrow array for each column.
```

Comment:
Not per se calculated. Something like this?

"method is a cheap operation as the validity bitmap can be read as a `Boolean` array."

Reply:
I guess I wanted to imply that not only can they be read as Booleans but that they already exist to be read - is that fair?